docker/build.sh

```diff
@@ -23,45 +28,115 @@ then
+                                        NVIDIA_BUILD_ID=${NVIDIA_BUILD_ID:-0}         \
+                                        GIT_SHA=${GIT_SHA}			      \
+                                        DALI_TIMESTAMP=${DALI_TIMESTAMP}	      \
+                                        NVIDIA_DALI_BUILD_FLAVOR=${NVIDIA_DALI_BUILD_FLAVOR} \
+                                        /opt/dali/build.sh && \
+                                        rm -rf /opt/dali/${DALI_BUILD_DIR}/nvidia* && \
+                                        cp /wheelhouse/* ./"
 
-if [ "$CREATE_RUNNER" = "YES" ]; then
-    echo "Runner image:" ${RUN_IMG}
-    echo "You can run this image with DALI installed inside, keep in mind to install neccessary FW package as well"
-    if [ ${CUDA_VERSION} == "9" ] ; then
-        export CUDA_IMAGE_NAME="nvidia/cuda:9.0-cudnn7-devel-ubuntu16.04"
-    elif [ ${CUDA_VERSION} == "10" ] ; then
-        export CUDA_IMAGE_NAME="nvidia/cuda:10.0-cudnn7-devel-ubuntu16.04"
-    else
-        echo "**************************************************************"
-        echo "Not supported CUDA version"
-        echo "**************************************************************"
+    if [ "$CREATE_RUNNER" = "YES" ]; then
+        echo "Runner image:" ${RUN_IMG}
+        echo "You can run this image with DALI installed inside, keep in mind to install neccessary FW package as well"
+        if [ ${CUDA_VERSION} == "9" ] ; then
+            export CUDA_IMAGE_NAME="nvidia/cuda:9.0-cudnn7-devel-ubuntu16.04"
+        elif [ ${CUDA_VERSION} == "10" ] ; then
+            export CUDA_IMAGE_NAME="nvidia/cuda:10.0-cudnn7-devel-ubuntu16.04"
+        else
+            echo "**************************************************************"
+            echo "Not supported CUDA version"
+            echo "**************************************************************"
+        fi
+        echo ${CUDA_VERSION}
+        echo ${CUDA_IMAGE_NAME}
+        export BUILDER_TMP=${BUILDER}_tmp
+        DOKCER_FILE_TMP=$(mktemp)
+        echo -e "FROM scratch\n"          \
+                "COPY ./${DALI_BUILD_DIR}/nvidia* /wheelhouse/\n" > ${DOKCER_FILE_TMP}
+        docker build -t ${BUILDER_TMP} -f ${DOKCER_FILE_TMP} .
+        docker build -t ${RUN_IMG} --build-arg "BUILD_IMAGE_NAME=${BUILDER_TMP}" --build-arg "CUDA_IMAGE_NAME=${CUDA_IMAGE_NAME}" --build-arg "PYVER=${PYVER}" --build-arg "PYV=${PYV}" -f Docker_run_cuda .
+        docker rmi ${BUILDER_TMP}
+        rm ${DOKCER_FILE_TMP}
     fi
-    echo ${CUDA_VERSION}
-    echo ${CUDA_IMAGE_NAME}
 
-    docker build -t ${RUN_IMG} --build-arg "BUILD_IMAGE_NAME=${BUILDER}" --build-arg "CUDA_IMAGE_NAME=${CUDA_IMAGE_NAME}" --build-arg "PYVER=${PYVER}" --build-arg "PYV=${PYV}" -f Docker_run_cuda .
-fi
+    if [ "$CREATE_WHL" = "YES" ]; then
+        mkdir -p ./wheelhouse
+        cp $(pwd)/${DALI_BUILD_DIR}/nvidia* ./wheelhouse/
+    fi
+else
+    echo "Build image:" ${BUILDER_WHL}
```

Comment:
If you do the `CMD` with BUILDER, you can do similar as in CI,
```
from BUILDER

COPY . /dali
```
build & run it here, and it will work on the copy instead of the mapping, using the last registered CMD.

Reply:
Will CMD create a layer inside the image with whl we can copy out?